lib/src/browser/web_compiler.dart

```diff
@@ -0,0 +1,15 @@
+@JS()
+library web_compiler;
+
+import 'package:js/js.dart';
+
+@JS(r'$dartLoader')
+external dynamic get _dartLoader;
+
+// The compiler used to compile the Dart to JS, determined at runtime
+// via a global window property that is only present when DDC is used.
+// If false, dart2js was used. Works with Dart 2 only.
+// Per the Dart team, the presence of this window property isn't
+// guaranteed forever, so proceed with caution and only use if you
+// absolutely MUST to work around compiler bugs.
+bool get isCompiledWithDdc => _dartLoader != null;
```

Comment:
I'd vote for making that a `///...` doc comment. If we do that, let's make the first sentence a short summary that is a [standalone paragraph](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph).